src/main/java/Main.java

```diff
@@ -8,6 +11,26 @@ public static void main(String[] args) {
     }
 
     public void run(Scanner scanner) {
-        // napisz swój program tutaj. Do wczytywania danych użyj przekazanego w parametrze scannera
+        boolean correctNumber = true;
+        List<Integer> numbers = new ArrayList<>();
+
+        while (correctNumber) {
+            System.out.println("Podaj liczbę całkowitą dodatnią: ");
+            int nextNumber = scanner.nextInt();
+
+            if (nextNumber > 0) {
+                numbers.add(nextNumber);
+            } else {
+                correctNumber = false;
+            }
+        }
+        try {
+            ListUtils.reverseList(numbers);
+            ListUtils.printSum(numbers);
+            ListUtils.printMin(numbers);
+            ListUtils.printMax(numbers);
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
```

Comment:
Unikaj sterowania programem wyjątkami, jest to używane jako antywzorzec: https://medium.com/@kritika.singhal/coding-by-exception-682060d8f0f1
W tym przypadku możnesz zamiast try dodać ifa sprawdzającego czy lista niej jest pusta.